src/device/pci/xhci.rs

```
@@ -462,7 +469,12 @@ impl XhciController {
                 // reads on the control endpoint), so we never reach this point
                 // when no device is available (expect for an invalid doorbell
                 // write, in which case panicking is the right thing to do.
-                self.real_device.as_mut().unwrap().transfer(ep as u8);
+                if slot_id <= self.device_slots.len() as u8 {
```

This function is only called by our internal logic and the slot is provided from our calculation based on doorbell address. Please make this an `assert(slot_id < MAX_SLOTS, "invalid doorbell");` and/or just dereference the device slot. If that fails (we don't have a device in that slot), we currently want to crash instead of continue silently without triggering the transfer.